admin/code/CampaignAdmin.php

```diff
@@ -53,19 +53,17 @@ class CampaignAdmin extends LeftAndMain implements PermissionProvider {
 	private static $thumbnail_height = 64;
 
 	public function getClientConfig() {
-		$urlSegment = Config::inst()->get($this->class, 'url_segment');
-
 		return array_merge(parent::getClientConfig(), [
 			'forms' => [
 				// TODO Use schemaUrl instead
 				'editForm' => [
 					'schemaUrl' => $this->Link('schema/EditForm')
 				]
 			],
-			'campaignViewRoute' => $urlSegment . '/:type?/:id?/:view?',
-			'itemListViewEndpoint' => $this->Link('set/:id/show'),
+			'campaignViewRoute' => $this->Link() . ':type?/:id?/:view?',
+			'itemListViewEndpoint' => $this->Link() . 'set/:id/show',
 			'publishEndpoint' => [
-				'url' => $this->Link('set/:id/publish'),
+				'url' => $this->Link() . 'set/:id/publish',
```

Comment:
The problem with this is that anything with a / in it isn't actually an action; ? can also get mangled my Controller::join_links, since it looks like a querystring delimiter.